sycl/test-e2e/VirtualMem/test-plan.md

```diff
@@ -0,0 +1,140 @@
+# Test plan for [`sycl_ext_oneapi_virtual_mem`][spec-link] extension
+
+## Testing scope
+
+### Device coverage
+
+Functionality provided by the extension is optional and it is only available if
+corresponding aspect (`ext_oneapi_virtual_mem`) is supported by a device.
+
+However, we still need to check that that we return a proper exception (with
+`feature_not_supported` error code) if the extension is attempted to be used on
+a device which doesn't support corresponding aspect.
+
+Therefore, tests should be structured in a way that negative testing is
+performed on all available devices, while positive tests can be skipped if
+device doesn't support the extension.
+
+### Type coverage
+
+APIs for physical and virtual memory allocation are not dependent on types and
+operate on opaque `void *` and `uintptr_t`. Therefore, there is no need for
+tests to be dependent on data types and to be repeated for every different data
+type there is.
+
+## Tests
+
+### Unit-tests
+
+Tests in this category may not perform some useful actions to exercise the
+extension functionality in full, but instead they are focused on making sure
+that all APIs are consistent with respect to other APIs.
+
+#### Memory granularity
+
+The following checks should be performed for `get_mem_granularity` API:
+- the returned value is always bigger than 0
+- recommended granularity is not smaller than minimum granularity
+- exception with `feature_not_supported` error code is thrown when `context`
+  passed to it contains multiple devices and not every of them have
+  `ext_oneapi_virtual_mem` aspect
+- **TODO**: do we need to check that returned granularity is not bigger than
+  maximum allocation size on a device?
```

Comment:
Interesting question. It's not necessarily guaranteed by the spec, but maybe it would be good to make sure. I would assume the max allocation still applies to these.